core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,161 @@
+    const chainId = isFork ? 1 : parseInt(getChainIdByNetworkType(network) || '', 16);
+    if (!isFork && !SUPPORTED_1INCH_NETWORK_IDS.includes(chainId)) {
+        throw new Error(`1inch does not support network ${network}`);
+    }
+    if (Number.isNaN(chainId)) {
+        throw new Error(`Invalid chainId: ${chainId}`);
+    }
+    const toTokenAddress = await getTokenAddressByNetworkAndSymbol(network, 'DAI');
+    const fromTokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateralSymbol);
+    const calleeAddress = getCalleeAddressByCollateralType(
+        network,
+        getCollateralConfigBySymbol(collateralSymbol).ilk,
+        marketId
+    );
+    // Documentation https://docs.1inch.io/docs/aggregation-protocol/api/swap-params/
+    const swapParams = {
+        fromTokenAddress,
+        toTokenAddress,
+        fromAddress: calleeAddress,
+        amount,
+        slippage,
+        allowPartialFill: false, // disable partial fill
+        disableEstimate: true, // disable eth_estimateGas
+        compatibilityMode: true, // always receive parameters for the `swap` call
+    };
+    const oneinchResponse = await executeOneInchApiRequest(chainId, '/swap', swapParams);
+    const functionSignature = ethers.utils.hexDataSlice(oneinchResponse.tx.data, 0, 4); // see https://docs.soliditylang.org/en/develop/abi-spec.html#function-selector
+    if (functionSignature !== EXPECTED_SIGNATURE) {
+        throw new Error(`Unexpected 1inch function signature: ${functionSignature}, expected: ${EXPECTED_SIGNATURE}`);
+    }
+    return oneinchResponse;
+}
+
+export async function extractPathFromSwapResponseProtocols(
+    network: string,
+    oneInchRoutes: OneInchSwapRoute[]
+): Promise<string[]> {
+    const pathStepsResolves = await Promise.all(
+        oneInchRoutes[0].map(async route => {
+            return await Promise.all([
+                await getErc20SymbolByAddress(network, route[0].fromTokenAddress),
+                await getErc20SymbolByAddress(network, route[0].toTokenAddress),
+            ]);
+        })
+    );
+    const path = [pathStepsResolves[0][0]];
+    for (const step of pathStepsResolves) {
+        path.push(step[1]);
+    }
+    return path;
+}
+
+export async function getOneInchMarketData(
+    network: string,
+    collateral: CollateralConfig,
+    amount: BigNumber,
+    marketId: string
+) {
+    const swapData = await getOneinchSwapParameters(
+        network,
+        collateral.symbol,
+        amount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(0),
+        marketId
+    );
+    const path = await extractPathFromSwapResponseProtocols(network, swapData.protocols);
+    const calleeData = swapData.tx.data;
+    const estimatedGas = swapData.tx.gas;
+    const exchangeFeeEth = new BigNumber(swapData.tx.gasPrice).multipliedBy(estimatedGas);
+    const exchangeFeeDai = new BigNumber(10).multipliedBy(exchangeFeeEth); // await convertETHtoDAI(network, exchangeFeeEth); TODO: uncomment
```

Comment:
Please use actual conversion

Reply:
hehehehe, this will result into recursion actually. 

here's the call tree:

- `convertEthToDai`
  - `getMarketPrice`
    - `getMarketDataRecords`
      - `getMarketDataById`
        - `if OneInchCallee`: `getOneInchMarketData`
          - `convertEthToDai`
            - ...

Reply:
As discussed in the daily, the suggestion is to use `convertSymbolToDai` (from `uniswapV3` helpers) for gas conversions (since gas prices doesn't need to be exchanged using the best market)

You can even further optimise it
- By calling it without amount parameter like `convertETHtoDAI` does (to not send request to uniswap for each specific gas amount, but rather do it once for 1 ETH, and cache it for all other gas prices)
- Update `convertETHtoDAI` to use the same structure, avoiding `getMarketPrice`